src/parser/shadowlands/modules/items/dungeons/InscrutableQuantumDevice.tsx

```diff
@@ -138,10 +143,8 @@ class InscrutableQuantumDevice extends Analyzer {
       Events.cast.by(SELECTED_PLAYER).spell(IQD_MANA),
       () => (this.counts.mana += 1),
     );
-    this.addEventListener(
-      // This is speculative as I could not find any log where the decoy had triggered
-      Events.cast.by(SELECTED_PLAYER).spell(IQD_DECOY),
-      () => (this.counts.decoy += 1),
+    this.addEventListener(Events.cast.by(SELECTED_PLAYER_PET).spell(IQD_DECOY_DISTRACT), (e) =>
```

Comment:
I've found in other code that mentions they can't use `by(SELECTED_PLAYER_PET)` for temporarily summoned pets, (like [here](https://github.com/oBusk/WoWAnalyzer/blob/c6c1f84eaaf76076aa308ca4feb965fc94859e09/analysis/deathknightblood/src/modules/shadowlands/conduits/WitheringPlague.tsx#L40-L41)) but it seems to work in this case? Is this something that works most of the time but not all of the time?

Reply:
possibly. or something that was fixed?